### PR TITLE
Validator blacklisting by hotkey [Step 1, Version 1]

### PR DESCRIPTION
**Pull request by dsavenkov1982**
Here are the changes related to the first step: custom config, blacklisting, blacklisting functionality, unit tests

### CODE REVIEW

File: neurons/miners/miner.py [hunk 1]
```diff
@@ -77,6 +78,7 @@ def get_config():
 
 
 def main(config):
+    print("Current Working Directory:", os.getcwd())
```
Comment (aph5nt) [inline neurons/miners/miner.py:81]: use bt.logging.debug instead of print, and remove Current Working Direcottory from logging

**Review by aph5nt — CHANGES_REQUESTED** [review]
some cleanup is needed

File: neurons/miners/miner.py (hunk 1 above)
Comment (dsavenkov1982) [reply neurons/miners/miner.py]: Addressed

File: neurons/miners/custom_miner_config.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            print(f"Error decoding JSON: {e}")
+            raise
+        except FileNotFoundError as e:
+            print(f"The file {self.json_file} was not found: {e}")
+            raise
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            raise
+
+# Example usage
```
[the remark this answers is not in the document]
Comment (dsavenkov1982) [reply neurons/miners/custom_miner_config.py]: Addressed

File: neurons/miners/blacklists.py [hunk 3]
```diff
@@ -5,16 +5,24 @@
...
 class BlacklistDiscovery:
-    def __init__(self, miner_config: MinerConfig, registry_manager: BlacklistRegistryManager):
+    def __init__(self, miner_config: MinerConfig, custom_miner_config: CustomMinerConfig, registry_manager: BlacklistRegistryManager):
         self.blacklist_registry_manager = registry_manager
         self.miner_config = miner_config
+        self.custom_miner_config = custom_miner_config
 
     def blacklist_discovery(self, metagraph, synapse: protocol.MinerDiscovery) -> typing.Tuple[bool, str]:
         hotkey = synapse.dendrite.hotkey
+        bt.logging.info(f"Discovered 777")
```
[the remark this answers is not in the document]
Comment (dsavenkov1982) [reply neurons/miners/blacklists.py]: Addressed

**Pull request closed without merging**